elastic/eui
# [EuiCodeBlock] Fix line numbers appearing in copied content

## Summary

I broke this behavior in https://github.com/elastic/eui/pull/6580 (annotation feature) by switching from rendering the line numbers as `::before` pseudo elements to actual text node. I've reverted the rendering to use pseudo elements, and cleaned up line number classNames slightly to better reflect the intended DOM structure (see individual commits).

I've additionally added new Cypress tests to ensure copy functionality doesn't regress in the future.

## QA

- Go to http://localhost:8030/#/editors-syntax/code#line-numbers
- Scroll down to the 2nd example with annotations
- Click the copy button, and paste into any text editor
-[ ] Confirm the code block content pastes without any line numbers present

### General checklist


~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
